contracts/stellar-interchain-token/src/contract.rs

```diff
@@ -81,7 +83,7 @@ impl StellarAssetInterface for InterchainToken {
 
         Self::receive_balance(&env, to.clone(), amount);
 
-        TokenUtils::new(&env).events().mint(owner, to, amount);
+        MintedEvent { to, amount }.emit(&env);
```

Comment:
the new events don't match the old ones so this is a breaking change. Consider why it was being done this way first however. We're using the library provided events for tokens to adhere to the standard token implementation, similar to what ERC20 tokens do. So we shouldn't deviate here. The event definition should match instead with the same fields and topics
https://docs.rs/soroban-token-sdk/latest/src/soroban_token_sdk/event.rs.html#25-28